src/bun.js/net.exports.js

```diff
@@ -368,7 +380,206 @@ export function createConnection(port, host, connectListener) {
 
 export const connect = createConnection;
 
+class Server extends EventEmitter {
+  #connectionListener;
+  #options;
+  #server;
+  #listening;
+  #connections;
+
+  constructor(options, connectionListener) {
+    super();
+    this.maxConnections = 0;
+
+    if (typeof options === "function") {
+      connectionListener = options;
+      options = {};
+    } else if (options == null || typeof options === "object") {
+      options = { ...options };
+    } else {
+      throw new Error("bun-net-polyfill: invalid arguments");
+    }
+
+    if (typeof options.maxConnections === "number" && options.maxConnections > 0) {
+      this.maxConnections = options.maxConnections;
+    }
+
+    this.#listening = false;
+    this.#connections = 0;
+    this.#connectionListener = connectionListener;
+    this.#options = options;
+  }
+
+  ref() {
+    this.#server?.ref();
+    return this;
+  }
+
+  unref() {
+    this.#server?.unref();
+    return this;
+  }
+
+  close(callback) {
+    if (this.#server) {
+      this.#server.stop(true);
+      this.#server = null;
+      this.#listening = false;
+      this.#connections = 0;
+      this.emit("close");
+      if (typeof callback === "function") {
+        callback();
+      }
+
+      return this;
+    }
+
+    if (typeof callback === "function") {
+      callback(new Error("Server is not open"));
+    }
+    return this;
+  }
+
+  address() {
+    if (this.#server) {
+      //TODO: fix adress when host is passed
+      let address = this.#server.hostname;
+      const type = isIP(address);
+      const port = this.#server.port;
+      if (typeof port === "number") {
+        return {
+          port,
+          address,
+          family: type ? `IPv${type}` : undefined,
+        };
+      }
+
+      if (type) {
+        return {
+          address,
+          family: type ? `IPv${type}` : undefined,
+        };
+      }
+
+      return address;
+    }
+    return null;
+  }
+
+  getConnections(callback) {
+    if (typeof callback === "function") {
+      callback(this.#server ? null : new Error("Server is not open"), this.#connections);
```

Comment:
does this error need a `.code`?

Reply:
Good catch i forgot about that I will get the right exception code from nodejs

Reply:
on `getConnections` actually we will never error (see the comment on last commit) https://github.com/oven-sh/bun/pull/2337/commits/dc6bc3c6d8d73d3f5588116eeeaab409d01c765f but I fixed the error code on `close`